.scrutinizer.yml

```diff
@@ -1,69 +1,11 @@
-inherit: true
+checks: 
+    php: true
 
-checks:
-  php:
-    verify_property_names: true
-    verify_argument_usable_as_reference: true
-    verify_access_scope_valid: true
-    useless_calls: true
-    use_statement_alias_conflict: true
-    variable_existence: true
-    unused_variables: true
-    unused_properties: true
-    unused_parameters: true
-    unused_methods: true
-    unreachable_code: true
-    too_many_arguments: true
-    sql_injection_vulnerabilities: true
-    simplify_boolean_return: true
-    side_effects_or_types: true
-    security_vulnerabilities: true
-    return_doc_comments: true
-    return_doc_comment_if_not_inferrable: true
-    require_scope_for_properties: true
-    require_scope_for_methods: true
-    require_php_tag_first: true
-    psr2_switch_declaration: true
-    psr2_class_declaration: true
-    property_assignments: true
-    prefer_while_loop_over_for_loop: true
-    precedence_mistakes: true
-    precedence_in_conditions: true
-    phpunit_assertions: true
-    php5_style_constructor: true
-    parse_doc_comments: true
-    parameter_non_unique: true
-    parameter_doc_comments: true
-    param_doc_comment_if_not_inferrable: true
-    optional_parameters_at_the_end: true
-    one_class_per_file: true
-    no_unnecessary_if: true
-    no_trailing_whitespace: true
-    no_property_on_interface: true
-    no_non_implemented_abstract_methods: true
-    no_error_suppression: true
-    no_duplicate_arguments: true
-    no_commented_out_code: true
-    newline_at_end_of_file: true
-    missing_arguments: true
-    method_calls_on_non_object: true
-    instanceof_class_exists: true
-    foreach_traversable: true
-    fix_line_ending: true
-    fix_doc_comments: true
-    duplication: true
-    deprecated_code_usage: true
-    deadlock_detection_in_loops: true
-    code_rating: true
-    closure_use_not_conflicting: true
-    catch_class_exists: false
-    blank_line_after_namespace_declaration: false
-    avoid_multiple_statements_on_same_line: true
-    avoid_duplicate_types: true
-    avoid_conflicting_incrementers: true
-    avoid_closing_tag: true
-    assignment_of_null_return: true
-    argument_type_checks: true
+build:
+    nodes:
+        analysis:
+            tests:
+                override: [php-scrutinizer-run]
 
 filter:
-  paths: [src/*]
+    paths: ["src/*", "tests/*"]

```

Comment:
Can you exclude tests/mock? Those pretty much only contain default SilverStripe definitions, so Scrutinizer will consistently complain about unused code because it doesn't understand SilverStripe